src/WebJobs.Script/Binding/DocumentDBScriptBindingProvider.cs

```diff
@@ -89,25 +104,59 @@ public override Collection<Attribute> GetAttributes()
 
                 string databaseName = Context.GetMetadataValue<string>("databaseName");
                 string collectionName = Context.GetMetadataValue<string>("collectionName");
+                string connection = Context.GetMetadataValue<string>("connection");
 
-                DocumentDBAttribute attribute = null;
-                if (!string.IsNullOrEmpty(databaseName) || !string.IsNullOrEmpty(collectionName))
+                if (Context.IsTrigger)
                 {
-                    attribute = new DocumentDBAttribute(databaseName, collectionName);
+                    CosmosDBTriggerAttribute attributeTrigger = new CosmosDBTriggerAttribute(databaseName, collectionName);
+                    attributeTrigger.ConnectionString = connection;
+
+                    JObject leasesConfiguration = Context.GetMetadataValue<JObject>("lease");
+                    if (leasesConfiguration != null)
+                    {
+                        JToken leaseCollectionName = leasesConfiguration["collectionName"];
+                        JToken leaseDatabaseName = leasesConfiguration["databaseName"];
+                        JToken leaseConnectionString = leasesConfiguration["connection"];
+
+                        if (leaseCollectionName != null)
+                        {
+                            attributeTrigger.LeaseCollectionName = leaseCollectionName.Value<string>();
+                        }
+
+                        if (leaseDatabaseName != null)
+                        {
+                            attributeTrigger.LeaseDatabaseName = leaseDatabaseName.Value<string>();
+                        }
+
+                        if (leaseConnectionString != null)
+                        {
+                            attributeTrigger.LeaseConnectionString = leaseConnectionString.Value<string>();
+                        }
+                    }
+
```

Comment:
I could not make the Lease Options work on the function level. Mainly because, even if I parse it correctly, I cannot pass it to the Attribute because it's not a basic Type, so it doesn't work in:

[CosmosDBTrigger("ItemDb", "ItemCollection", LeaseOptions= new LeaseOptions(...))]